src/Polly/Utilities/KeyHelper.cs

```diff
@@ -6,5 +6,9 @@ internal static class KeyHelper
     private const int GuidPartLength = 8;
 
     public static string GuidPart() =>
+#if NET6_0_OR_GREATER
+        Guid.NewGuid().ToString()[..GuidPartLength];
```

Comment:
Isn't this just adding extra complexity?

Reply:
Is it more code? Yes. Is it complicated? No, it's in two small places.

It fails the build with the .NET 9 SDK as it wasn't smart enough to detect it in .NET 8, so we need to do _something_.

Alternatively, I can just suppress the warning and put it back how it was.

Reply:
no problem then, thanks for explanation